### PR TITLE
[#1180] Update gtag and google tag manager

### LINKED ISSUE #1180: Google Analytics does not measure form steps correctly
Currently, Google Analytics measures views and pages but all forms are opened in the same page (called "Openforms").

We need to change the title of the window to match the form, and maybe even include the step, to get this working. This also makes the window title show nicely, so win-win.

### PR DESCRIPTION
**Pull request by SilviaAmAm**
Fixes #1180 

Front end PR: https://github.com/open-formulieren/open-forms-sdk/pull/269

### CODE REVIEW

File: src/openforms/templates/master.html [hunk 1]
```diff
@@ -29,6 +29,14 @@
 </head>
 
 <body class="view__body">
+    {% if enable_analytics and analytics_tools_config.gtm_code %}
```
Comment (sergei-maertens) [inline src/openforms/templates/master.html:32]: shouldn't this also only be emitted if Google Analyics is explicitly enabled - not only based on whether the code is filled out or not?

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

File: src/openforms/templates/master.html (hunk 1 above)
Comment (SilviaAmAm) [reply src/openforms/templates/master.html:32]: This snippet should be present when the other Google Tag Manager script is present (https://github.com/open-formulieren/open-forms/blob/master/src/openforms/analytics_tools/templates/analytics_tools/google.html#L8). Unless i read it wrong, I think the other one relies on `enable_analytics` and `analytics_tools_config.gtm_code`?

Comment (sergei-maertens) [reply src/openforms/templates/master.html:32]: I see that the model has the field `enable_google_analytics`, but this is not used anywhere now - this should be used similar to `{% if analytics_tools_config.piwik_enabled %}` in `src/openforms/analytics_tools/templates/analytics_tools/piwik.html`

The idea is that you can configure your integration without enabling it.

File: src/openforms/templates/master.html [hunk 2]
```diff
@@ -29,6 +29,14 @@
 </head>
 
 <body class="view__body">
+    {% if enable_analytics and analytics_tools_config.enable_google_analytics and analytics_tools_config.gtm_code %}
+        {# Load the second part of the Google tag manager snippet #}
+        <!-- Google Tag Manager (noscript) -->
+        <noscript><iframe src="https://www.googletagmanager.com/ns.html?id={{ analytics_tools_config.gtm_code }}"
+        height="0" width="0" style="display:none;visibility:hidden"></iframe></noscript>
```
Comment (sergei-maertens) [inline src/openforms/templates/master.html:36]: I think this inline style will be blocked by the CSP

**Review by sergei-maertens — CHANGES_REQUESTED** [review]

File: docs/developers/csp.rst [hunk 3]
```diff
@@ -84,6 +84,11 @@ do not rely on external CDNs (such as the Formio CDN, Google Fonts...).
 
 * The ``img-src`` requires ``data:``, otherwise map and signature components break.
 
+* If Google Tag manager is enabled, the ``img-src`` requires ``www.googletagmanager.com`` to allow the execution of the
+  Tag Manager container code. More information can be found `here`_.
+
+.. _here: https://developers.google.com/tag-platform/tag-manager/web/csp#enable_the_container_tag_to_use_csp
+
```
Comment (sergei-maertens) [inline docs/developers/csp.rst:91]: This should be added to the automatic CSP configuration when activating/deactivating an analytics tool: `src/openforms/analytics_tools/contrib/google_analytics/csp_headers.json`

File: src/openforms/templates/master.html [hunk 4]
```diff
@@ -29,6 +29,14 @@
 </head>
 
 <body class="view__body">
+    {% if enable_analytics and analytics_tools_config.enable_google_analytics and analytics_tools_config.gtm_code %}
+        {# Load the second part of the Google tag manager snippet #}
+        <!-- Google Tag Manager (noscript) -->
+        <noscript><iframe class="google-tag-manager__noscript" src="https://www.googletagmanager.com/ns.html?id={{ analytics_tools_config.gtm_code }}"
```
Comment (sergei-maertens) [inline src/openforms/templates/master.html]: this is not correct BEM naming - an element cannot exist without a parent block (there's no `.google-tag-manager` class anywhere)

File: src/openforms/scss/components/admin/_google-tag-manager.scss [hunk 5]
```diff
@@ -0,0 +1,6 @@
+.google-tag-manager {
```
Comment (sergei-maertens) [inline src/openforms/scss/components/admin/_google-tag-manager.scss]: not the correct location - this is not an admin component, but a public component (it gets loaded on the form detail page)

**Review by sergei-maertens — APPROVED** [review]

**Pull request merged**